src/buffer/out/textBuffer.cpp

```diff
@@ -1028,3 +1028,243 @@ const TextBuffer::TextAndColor TextBuffer::GetTextForClipboard(const bool lineSe
 
     return data;
 }
+
+// Routine Description:
+// - Generates a CF_HTML compliant structure based on the passed in text and color data
+// Arguments:
+// - rows - the text and color data we will format & encapsulate
+// - iFontHeightPoints - the unscaled font height
+// - fontFaceName - the name of the font used
+// Return Value:
+// - string containing the generated HTML
+std::string TextBuffer::GenHTML(const TextAndColor& rows, const int iFontHeightPoints, const PCWCHAR fontFaceName)
+{
+    std::string szClipboard; // we will build the data going back in this string buffer
+
+    try
+    {
+        std::string const szHtmlClipFormat =
+            "Version:0.9\r\n"
+            "StartHTML:%010d\r\n"
+            "EndHTML:%010d\r\n"
+            "StartFragment:%010d\r\n"
+            "EndFragment:%010d\r\n"
+            "StartSelection:%010d\r\n"
+            "EndSelection:%010d\r\n";
+
+        // measure clip header
+        size_t const cbHeader = 157; // when formats are expanded, there will be 157 bytes in the header.
+
+        std::string const szHtmlHeader =
+            "<!DOCTYPE><HTML><HEAD><TITLE>Windows Console Host</TITLE></HEAD><BODY>";
+        size_t const cbHtmlHeader = szHtmlHeader.size();
+
+        std::string const szHtmlFragStart = "<!--StartFragment -->";
+        std::string const szHtmlFragEnd = "<!--EndFragment -->";
+        std::string const szHtmlFooter = "</BODY></HTML>";
+        size_t const cbHtmlFooter = szHtmlFooter.size();
+
+        std::string const szDivOuterBackgroundPattern = R"X(<DIV STYLE="background-color:#%02x%02x%02x;white-space:pre;">)X";
+
+        size_t const cbDivOuter = 55;
+        std::string szDivOuter;
+        szDivOuter.reserve(cbDivOuter);
+
+        std::string const szSpanFontSizePattern = R"X(<SPAN STYLE="font-size: %dpt">)X";
+
+        size_t const cbSpanFontSize = 28 + (iFontHeightPoints / 10) + 1;
+
+        std::string szSpanFontSize;
+        szSpanFontSize.resize(cbSpanFontSize + 1); // reserve space for null after string for sprintf
+        sprintf_s(szSpanFontSize.data(), cbSpanFontSize + 1, szSpanFontSizePattern.data(), iFontHeightPoints);
+        szSpanFontSize.resize(cbSpanFontSize); //chop off null at end

```

Comment:
nitpick: This might be the only way to do formatting, but the resize, format, resize again seems cumbersome.

I know everybody hates C++ streams but it seems that they would work relatively well for this function at least.